### PR TITLE
build01: switch to zfs

### PR DESCRIPTION
**Pull request by zowoq**
Had issues with mirrored boot so not going to bother with it for now.

@Mic92 This config passes the nixos-anywhere vm-test but can you take a look to make sure the zfs stuff is okay?

### CODE REVIEW

File: modules/nixos/disko-zfs.nix [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    zpool = {
+      zroot = {
+        type = "zpool";
+        options = {
+          ashift = "12";
+        };
+        rootFsOptions = {
+          acltype = "posixacl";
+          atime = "off";
+          compression = "off";
```
Comment (Mic92) [inline modules/nixos/disko-zfs.nix:74]: Wouldn't lz4 makes sense here?

```suggestion
          compression = "lz4";
```

File: modules/nixos/common/default.nix [hunk 2]
```diff
@@ -29,7 +29,7 @@
 
   srvos.flake = inputs.self;
 
-  boot.kernelPackages = pkgs.linuxPackages_latest;
+  boot.kernelPackages = pkgs.lib.mkIf (!config.boot.supportedFilesystems.zfs or false) pkgs.linuxPackages_latest;
```
Comment (Mic92) [inline modules/nixos/common/default.nix:32]:
```suggestion
  boot.kernelPackages = if config.boot.supportedFilesystems.zfs or false then pkgs.zfsUnstable.latestCompatibleLinuxPackages else pkgs.linuxPackages_latest;
```

Comment (zowoq) [reply modules/nixos/common/default.nix:32]: Following the stable kernel is intentional, don't want kernel versions downgrades, e.g. in our next nixpkgs bump it'll go from 6.7 -> 6.6.

**Comment by zowoq** [comment]
I wasn't planning to put zfs on build01 but seemed like a good opportunity to test this config.

**Pull request merged**